core/src/main/java/com/sequenceiq/cloudbreak/converter/ClusterToJsonConverter.java

```diff
@@ -167,6 +169,11 @@ private void convertComponentConfig(ClusterResponse response, Long clusterId) {
                 AmbariRepoDetailsJson ambariRepoDetailsJson = conversionService.convert(ambariRepo, AmbariRepoDetailsJson.class);
                 response.setAmbariRepoDetailsJson(ambariRepoDetailsJson);
             }
+            StackRepoDetails stackRepoDetails = componentConfigProvider.getHDPRepo(clusterId);
```

Comment:
Can source.getComponents() be used here? We can avoid an unnecessary DB call.

Reply:
ok @keyki I will refactor that part as well